File: 1021 - Remove Outermost Parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string res;
        int d = 0;
        for (char c : s) {
            if (c == ')') d--;
            if (d > 0) res += c;
            if (c == '(') d++;
        }
        return res;
    }
};

File: 1021 - Remove Outermost Parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        res = ''
        d = 0
        for c in s:
            if c == ')':
                d -= 1
            if d >= 1:
                res += c
            if c == '(':
                d += 1
        return res